### PR TITLE
Run XManager Vizier hyperparameter sweep on GPU

### PR DESCRIPTION
**Pull request by panford**
## Description
This feature enables XManager-Vertex AI vizier hyperparameter sweep done faster on GPUs.

## Types of changes
New docker images created from tensorflow-gpu base image to enable training on GPUs.

## What changes does to code
Adds new feature:- XManager + Vertex AI + Vizier on GPUs

### CODE REVIEW

File: src/skai/model/xmanager_external_metric_logger.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
 from typing import Union
-
-import tensorflow as tf
 from log_metrics_callback import MetricLogger
-from xmanager.vizier.vizier_cloud.vizier_worker import \
-    VizierWorker  # Open-source manager-Deepmind
+from xmanager_helper_functions import VizierWorker
+# from xmanager.vizier.vizier_cloud.vizier_worker import \
+#     VizierWorker  # Open-source manager-Deepmind
```
Comment (Alikerin) [inline src/skai/model/xmanager_external_metric_logger.py:7]: can we remove these commented lines?

File: src/skai/model/xm_launch_single_model_vertex.py [hunk 2]
```diff
@@ -12,29 +12,83 @@
...
+        "WORKDIR /skai",
         "COPY skai/requirements.txt /skai/requirements.txt",
-        "RUN python -m pip install -r skai/requirements.txt",
+        "RUN pip3 install --upgrade pip",
+        "RUN pip3 install --timeout 1000 -r requirements.txt",
         "COPY skai/ /skai",
-        "RUN chown -R 1000:root /skai && chmod -R 775 /skai",
-        "WORKDIR /skai",
     ]
+    return base_image, docker_instructions
```
Comment (Alikerin) [inline src/skai/model/xm_launch_single_model_vertex.py:91]: Can we move these docker stuff to a different file?

File: src/skai/model/xm_launch_single_model_vertex.py [hunk 3]
```diff
@@ -12,29 +12,83 @@
...
+        "RUN chmod 700 /lib/libtpu.so",
+        "RUN wget "
+        "https://storage.googleapis.com/cloud-tpu-tpuvm-artifacts/tensorflow/tf-2.9.1/"
+        "tensorflow-2.9.1-cp38-cp38-linux_x86_64.whl",
+        "RUN pip3 install tensorflow-2.9.1-cp38-cp38-linux_x86_64.whl",
+        "RUN rm tensorflow-2.9.1-cp38-cp38-linux_x86_64.whl",
+    ]
+
+
+def construct_docker_instructions(accelerator: str) -> Tuple[str, List[str]]:
```
Comment (Alikerin) [inline src/skai/model/xm_launch_single_model_vertex.py:49]: Would it be better if we name this function as `get_docker_instructions`?

File: src/skai/model/xm_launch_single_model_vertex.py [hunk 4]
```diff
@@ -154,17 +208,18 @@ def main(_) -> None:
...
       )
   ) as experiment:
-
+    base_image, docker_instructions = construct_docker_instructions(FLAGS.accelerator)
     executable_spec = xm.PythonContainer(
         # Package the current directory that this script is in.
         path=os.path.expanduser(FLAGS.project_path),
-        base_image='gcr.io/deeplearning-platform-release/base-gpu',
-        docker_instructions=get_docker_instructions(),
+        base_image=base_image, #'gcr.io/deeplearning-platform-release/base-gpu',
```
Comment (Alikerin) [inline src/skai/model/xm_launch_single_model_vertex.py:221]: can you remove the commented image here?

**Review by Alikerin — CHANGES_REQUESTED** [review]
I left a few suggestions.
Please take a look.
Thank you.

**Comment by Alikerin** [comment]
In the description you mention Vizier and DataFlow, is this PR related to both?

**Comment by Alikerin** [comment]
Also try fix the conflicts with instadeep branch

File: src/skai/model/xm_launch_single_model_vertex.py (hunk 3 above)
Comment (panford) [reply src/skai/model/xm_launch_single_model_vertex.py:49]: yes, that makes more sense

**Review by Alikerin — APPROVED** [review]

**Review by Alikerin — APPROVED** [review]

**Pull request merged**